pkg/apis/events/v1/cloudstoragesource_types.go

```diff
@@ -0,0 +1,149 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	gcpduckv1 "github.com/google/knative-gcp/pkg/apis/duck/v1"
+	kngcpduck "github.com/google/knative-gcp/pkg/duck/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/kmeta"
+	"knative.dev/pkg/webhook/resourcesemantics"
+)
+
+// +genclient
+// +genreconciler
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CloudStorageSource is a specification for a Google Cloud CloudStorageSource Source resource
+type CloudStorageSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   CloudStorageSourceSpec   `json:"spec"`
+	Status CloudStorageSourceStatus `json:"status"`
+}
+
+// Verify that CloudStorageSource matches various duck types.
+var (
+	_ apis.Defaultable             = (*CloudStorageSource)(nil)
+	_ apis.Validatable             = (*CloudStorageSource)(nil)
+	_ kmeta.OwnerRefable           = (*CloudStorageSource)(nil)
+	_ resourcesemantics.GenericCRD = (*CloudStorageSource)(nil)
+	_ kngcpduck.Identifiable       = (*CloudStorageSource)(nil)
+	_ kngcpduck.PubSubable         = (*CloudStorageSource)(nil)
+	_ duckv1.KRShaped      = (*CloudStorageSource)(nil)
+)
+
+// CloudStorageSourceSpec is the spec for a CloudStorageSource resource
+type CloudStorageSourceSpec struct {
+	// This brings in the PubSub based Source Specs. Includes:
+	// Sink, CloudEventOverrides, Secret, PubSubSecret, and Project
+	gcpduckv1.PubSubSpec `json:",inline"`
+
+	// Bucket to subscribe to.
+	Bucket string `json:"bucket"`
+
+	// EventTypes to subscribe to. If unspecified, then subscribe to all events.
+	// +optional
+	EventTypes []string `json:"eventTypes,omitempty"`
+
+	// ObjectNamePrefix limits the notifications to objects with this prefix
+	// +optional
+	ObjectNamePrefix string `json:"objectNamePrefix,omitempty"`
+}
+
+const (
+	// CloudStorageSourceConditionReady has status True when the CloudStorageSource is ready to send events.
+	CloudStorageSourceConditionReady = apis.ConditionReady
+
+	// NotificationReady has status True when GCS has been configured properly to
+	// send Notification events
+	NotificationReady apis.ConditionType = "NotificationReady"
+)
+
+var storageCondSet = apis.NewLivingConditionSet(
+	gcpduckv1.PullSubscriptionReady,
+	gcpduckv1.TopicReady,
+	NotificationReady)
+
+// CloudStorageSourceStatus is the status for a GCS resource
+type CloudStorageSourceStatus struct {
+	// This brings in our GCP PubSub based events importers
```

Comment:
nit: This brings in the Status for our GCP PubSub event sources

Reply:
and add `.` at the end of comments in multiple places if you can. Sorry, we have been missing those in previous api versions. If we can have them for v1, it'd be great